flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableEnvironmentImpl.java

```diff
@@ -323,38 +324,38 @@ public void insertInto(Table table, String path, String... pathContinued) {
 
 	@Override
 	public void sqlUpdate(String stmt) {
-		List<Operation> operations = planner.parse(stmt);
-
-		if (operations.size() != 1) {
-			throw new TableException(
-				"Unsupported SQL query! sqlUpdate() only accepts a single SQL statement of type " +
-					"INSERT, CREATE TABLE, DROP TABLE");
-		}
+		Consumer<Operation> operationPreConsumer = operation -> {
```

Comment:
We do not need to push the register table logic into Planner#parse. The `parse` only parse a plain String into a structured `Operation`. The register logic should still exist in `TableEnvironmentImpl`.

Reply:
Thanks, @wuchong . If we leave register logic in TableEnvironmentImpl, `SqlToOperationConverter#convert` would throw a ValidationException when invokes `flinkPlanner.validate` to validate DML(like `insert into t2 select ... from t2`) SqlNode because the source or sink table is not found that time. So it seems we need to consume DDL operations in advance before converting DML SqlNode to Operation. What do you think is the better solution here?

Reply:
Thanks for the explanation. I get it. 
How about changing the return type of `Planner#parse` from `List<Operation>` to `Iterable<Operation>`? In this way, we can process the lazily produced Operation one by one. 

What do you think about this problem? @twalthr @dawidwys @godfreyhe

Reply:
> How about changing the return type of `Planner#parse` from `List<Operation>` to `Iterable<Operation>`? In this way, we can process the lazily produced Operation one by one.

Sounds good and thanks for the reminder! I will push another commit.

Reply:
Changed by [0b54672 ](https://github.com/apache/flink/pull/9781/commits/0b54672d7eb381ac2cc422dd2da14518fa4d70d2)